### PR TITLE
docs: add parameter specifications to user guide.

### PR DESCRIPTION
**Pull request by Zer0Legion**
Now, all the constraints of the different parameters are collated in the user guide, helping the user to follow the instructions better and increasing reader's comprehension. Use of Logos.

Lets go home

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -50,16 +50,18 @@ Avengers Assemble (AA) is a **desktop app for managing contacts, optimized for u
...
+| [t/]    | Tags of person             | Should be alphanumeric.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             |
+| [m/]... | Matriculation ID of person | The first letter must be an uppercase 'A', followed by 7 numbers, and ending with an uppercase letter.                                                                                                                                                                                                                                                                                                                                                                                                                                              |
```
Comment (delishad21) [inline docs/UserGuide.md:64]: are the specifications for tags and matric flipped? Each person should only have one matric ID while it can have multiple tags.

File: docs/UserGuide.md [hunk 2]
```diff
@@ -88,16 +90,6 @@ Shows a message explaning how to access the help page.
...
 
-
-### Parameters:
-
-* n/NAME
-* p/PHONE_NUMBER
-* e/EMAIL
-* a/ADDRESS
-* t/TAG (Optional)
-* m/MATRICULATION_NUMBER (Optional)
```
Comment (delishad21) [inline docs/UserGuide.md:99]: Seems like irrelevant information. Nice spot!

**Review by delishad21 — COMMENTED** [review]
Just pointing out some small things. Good change, the user guide is now more detailed and comprehensive.

File: docs/UserGuide.md [hunk 3]
```diff
@@ -50,16 +50,18 @@ Avengers Assemble (AA) is a **desktop app for managing contacts, optimized for u
...
-|p/      | Phone number of person    |
-|e/      | Email of person           |
-|a/      | Address of person         |
-|t/      | Tags of person            |
-|m/      | Matriculation ID of person|
+* Here are a list of valid prefixes and what they each refer to. 
+* Prefixes encased with '[ ]' are optional.
+* Prefixes with '…' after them can be used multiple times.
+
+| Prefix  | What it refers to          | Constraints                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         |
```
Comment (danielleloh) [inline docs/UserGuide.md:57]: good job adding constraints into the table as I think it improves the readability of the user guide and allows users to follow the user guide easier

File: docs/UserGuide.md (hunk 1 above)
Comment (jayllo-c) [reply docs/UserGuide.md]: Good catch pointing this potential issue out
